### PR TITLE
add code coverage prompt

### LINKED ISSUE #34: Add `.nyc_output` and `coverage` to gitignore

### PR DESCRIPTION
**Pull request by radiovisual**
I have added a question at the prompt:

`Do you need code coverage (nyc)?`

If the answer is `Yes`, the following occurs:

- Adds `.nyc_output` and `coverage` to the `.gitignore` file
- Appends `&& nyc ava` to the npm "test" script: `"test": "xo && ava && nyc ava"`
- Adds `"nyc"` to the `devDependencies` in `package.json`

I have also included a unit test for this behavior.

This PR would close issue #34 if it is merged.

### CODE REVIEW

**Comment by SamVerschueren** [comment]
```json
"test": "xo && ava && nyc ava"
```

should be 

```json
"test": "xo && nyc ava"
```

Not sure of adding this in the main generator though.

**Comment by radiovisual** [comment]
I was actually thinking about that. It would be nice to have the ability to add `nyc` anytime after the module has already been initialized using `$ nm:nyc`, this would encapsulate the nyc option in its own sub-generator, but also provide the nyc option at the prompt with the question:

`Do you need code coverage (nyc)?` 

Also, I initially missed the detail in the [comment made by @sindresorhus on issue #34](https://github.com/sindresorhus/generator-nm/issues/34#issuecomment-161930056) regarding the sub-generator, and my approach doesn't use sub-generators at the moment. 

I am happy to refactor this into a sub-generator if that is the popular vote.

**Comment by radiovisual** [comment]
...but then again, adding in a sub-generator __would__ increase the complexity of this generator. The way I have it implemented in this current PR, it only takes a few lines in the templates to get things working, but if we create an entire sub-generator, then suddenly the module doubles in size (because instead of simply embedding `<% if %>` statements in the templates, we would need to do read/write operations on two different files: `package.json`, and `.gitignore` and handle all the reporting/errors. So it's worth considering when weighing the costs of a sub-generator. I have already started working on a sub-generator branch just to get an idea of what it will involve. I am happy to share for comparison.

**Comment by radiovisual** [comment]
Ok, so I have created a generator called [generator-avanyc](https://github.com/radiovisual/generator-avanyc) which adds nyc automatically to any existing AVA project. So I can independently (or cooperatively) experiment with this as a sub-generator to see what the different experiences are like. 

I have typed a lot code, but my feelings won't be hurt if you decide my contributions don't fit the vision.

**_Free hamburgers!_** :point_right: :hamburger: :hamburger: :hamburger:

File: app/templates/_package.json [hunk 1]
```diff
@@ -30,6 +30,7 @@
   <% } %>},
   "devDependencies": {
     "ava": "^0.11.0",
-    "xo": "^0.12.1"
+    "xo": "^0.12.1"<% if (nyc) { %>,
+    "nyc": "^5.6.0"<% } %>
```
Comment (sindresorhus) [inline app/templates/_package.json]: Can you update this to latest `nyc` version?

**Comment by sindresorhus** [comment]
I'm gonna go with the simple solution. This looks good. Can you fix the merge conflict?

File: app/index.js [hunk 2]
```diff
@@ -31,6 +31,11 @@ module.exports = yeoman.Base.extend({
 			message: 'Do you need a CLI?',
 			type: 'confirm',
 			default: false
+		}, {
+			name: 'nyc',
+			message: 'Do you need code coverage (nyc)?',
```
Comment (sindresorhus) [inline app/index.js]: Remove `(nyc)`. It doesn't really add anything useful.

File: app/templates/_package.json (hunk 1 above)
Comment (radiovisual) [inline app/templates/_package.json]: Sure, you want to use:

```
"nyc": "*"
```
or 
```
"nyc": "^6.1.1"
```

?

**Comment by radiovisual** [comment]
merge conflict fixed.

**Comment by sindresorhus** [comment]
Thank you @radiovisual :)

**Comment by radiovisual** [comment]
No problem! Happy to help! :smile:

**Pull request closed without merging**